platform/src/components/aws/bucket.ts

```diff
@@ -620,34 +649,43 @@ export class Bucket extends Component implements Link.Linkable {
 
     function createBucketPolicy() {
       return access.apply((access) => {
-        const statements = [];
-        if (access) {
+        if (!access && !args.getAdditionalPolicyStatements) return;
+        const statements = args.getAdditionalPolicyStatements
+          ? args.getAdditionalPolicyStatements(bucket)
+          : [];
+        if (["public", "cloudfront"].includes(access ?? "")) {
           statements.push({
             principals: [
               access === "public"
                 ? { type: "*", identifiers: ["*"] }
                 : {
-                  type: "Service",
-                  identifiers: ["cloudfront.amazonaws.com"],
-                },
+                    type: "Service",
+                    identifiers: ["cloudfront.amazonaws.com"],
+                  },
             ],
             actions: ["s3:GetObject"],
             resources: [interpolate`${bucket.arn}/*`],
           });
         }
-        statements.push({
-          effect: "Deny",
-          principals: [{ type: "*", identifiers: ["*"] }],
-          actions: ["s3:*"],
-          resources: [bucket.arn, interpolate`${bucket.arn}/*`],
-          conditions: [
-            {
-              test: "Bool",
-              variable: "aws:SecureTransport",
-              values: ["false"],
-            },
-          ],
-        });
+        if (
+          ["public", "cloudfront", "denyUnsecureTransport"].includes(
+            access ?? "",
```

Comment:
there is an extra `,` should be removed. And make this if condition one line would be much easier to read.